### ISSUE TITLE
Build fails with "invalid reference to syscall.recvmsg" on Go 1.23

### ISSUE THREAD

**Issue by svoop** [acme-dns/acme-dns#365]
I'm trying to rebuild acme-dns on a Gentoo box after upping to Go 1.23 and it fails with:

```
link: golang.org/x/net/internal/socket: invalid reference to syscall.recvmsg
```

Apparently, access to internal symbols has changed with Go 1.23, at least that's what's [causing this error on other programs](https://github.com/chipmk/docker-mac-net-connect/issues/39#issuecomment-2299898790).

I'm not familiar with Go, so that's pretty much the end of the line for me. Any ideas how to rewrite the offending code not to use internal symbols?

**Comment by svoop** [comment]
@joohoi The trend for now is to pin Go to 1.22 as a workaround. Since I don't know Go, could you drop a line as to how likely it is to make acme-dns compatible with newer versions of Go at all? From the frequency of commits, you might have bigger fish to fry which I'd totally understand!

In my particular case, I could switch to a ACME plugin which directly updates recs on my DNS provider, but I'd have to write an OS package (Gentoo ebuild that is) for this plugin first. I'm more inclined to stay with acme-dns, but if you intend to sunset acme-dns e.g. because of the workload to keep up with newer versions of Go, I'd start working on the package.

Thanks a bunch for a small update on where you intend to go with acme-dns. And big thanks for all the work you've already done... from one FOSS author to another! 👍

**Comment by svoop** [comment]
@joohoi The missing compatibility with Go 1.23 starts to produce trouble with other software built with Go. Could you give a hint as to whether an update for Go 1.23 is realistic or not so folks like me can work around or move on? I'm a big fan of acme-dns, really like the way it solves the problems that come with DNS proof with Cerbot. 

I see that you didn't commit much recently which is totally okay of course. But please drop a word here on whether you plan to abandon acme-dns... which would be sad but totally okay as well. Just, we have to know.

Thanks a bunch!!

**Comment by joohoi** [comment]
Hi @svoop 

I made the required changes and pushed a PR for Go 1.23 support #368 

While it mentions the minimum required version of Go being 1.22, I tested it out with 1.23 as well. With the newest versions of all dependencies, 1.22 is the minimum required version though.

**Comment by thedix** [comment]
@joohoi Thanks for your work!
Are there any plans to merge it to master?

**Comment by joohoi** [comment]
Yeah, and to bump version to be sure. There seems to be an issue with the
linter, so I’ll have to fix that first though.

**Comment by svoop** [comment]
@joohoi Great, thanks so much!!

**Comment by hemna** [comment]
```└─> go install github.com/MickaelBergem/dnsstresss@latest
go: downloading github.com/MickaelBergem/dnsstresss v0.0.0-20231025101534-00dcd6dafb95
go: downloading github.com/logrusorgru/aurora v2.0.3+incompatible
go: downloading github.com/miekg/dns v1.1.31
go: downloading golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading golang.org/x/sys v0.0.0-20190924154521-2837fb4f24fe
# github.com/MickaelBergem/dnsstresss
link: golang.org/x/net/internal/socket: invalid reference to syscall.recvmsg

└─> go version
go version go1.24.6 darwin/arm64
```
